brainiak/utils/utils.py

```diff
@@ -350,3 +351,252 @@ def read_afni(self, fname):
+        In such cases, these parameters will default to 1.0.
+
+        'AFNI' style has one line for each scan (run).
+        Each line has a few triplets in the format of
+        stim_onsets*weight:duration
+        (or simpler, see below), separated by spaces.
+        For example, 3.2*2.0:1.5 means that one event starts at 3.2s,
+        modulated by weight of 2.0 and lasts for 1.5s.
+        If some run does not include a single event
+        of a condition (stimulus type), then you can put *,
+        or a negative number, or a very large number in that line.
+        Either duration or weight can be neglected. In such
+        cases, they will default to 1.0.
+        For example, 3.0, 3.0*1.0, 3.0:1.0 and 3.0*1.0:1.0 all
+        means an event starting at 3.0s, lasting for 1.0s, with
+        amplitude modulation of 1.0.
+
+
+    Returns
+    -------
+
+    design: 2D numpy array
+        design matrix. Each time row represents one TR
+        (fMRI sampling time point) and each column represents
+        one experiment condition, in the order in stimtime_files
+
+    """
+    if np.ndim(scan_duration) == 0:
+        scan_duration = [scan_duration]
+    scan_duration = np.array(scan_duration)
+    assert np.all(scan_duration > TR), \
+        'scan duration should be longer than a TR'
+    if type(stimtime_files) is str:
+        stimtime_files = [stimtime_files]
+    assert TR > 0, 'TR should be positive'
+    assert style == 'FSL' or style == 'AFNI', 'style can only be FSL or AFNI'
+    n_C = len(stimtime_files)  # number of conditions
+    n_S = np.size(scan_duration)  # number of scans
+    if n_S > 1:
+        design = [np.empty([int(np.floor(duration / TR)), n_C])
+                  for duration in scan_duration]
+    else:
+        design = [np.empty([int(np.floor(scan_duration / TR)), n_C])]
+    scan_onoff = np.insert(np.cumsum(scan_duration), 0, 0)
+    if style == 'FSL':
+        design_info = read_stimtime_FSL(stimtime_files, n_C, n_S, scan_onoff)
+    elif style == 'AFNI':
+        design_info = read_stimtime_AFNI(stimtime_files, n_C, n_S, scan_onoff,
+                                         scan_duration)
+
+    # generate design matrix
+    for i_s in range(n_S):
+        for i_c in range(n_C):
+            stimfunction = generate_stimfunction(
+                onsets=design_info[i_s][i_c]['onset'],
+                event_durations=design_info[i_s][i_c]['duration'],
+                total_time=scan_duration[i_s],
+                weights=design_info[i_s][i_c]['weight'])
+            design[i_s][:, i_c] = double_gamma_hrf(
+                stimfunction, TR, response_delay=6, undershoot_delay=12,
+                response_dispersion=0.9, undershoot_dispersion=0.9,
+                response_scale=1, undershoot_scale=0.035, scale_function=0)
+    return np.concatenate(design, axis=0)
+
+
+def read_stimtime_FSL(stimtime_files, n_C, n_S, scan_onoff):
+    """Utility called by gen_design. It reads in one or more
```

Comment:
If this is only called by `gen_design`, please make it private by prepending a `_`. Otherwise, do not start the docstring with a reference to `gen_design`.